tools/orca/src/model.rs

```diff
@@ -84,6 +124,96 @@ pub async fn login(
     }
 }
 
+pub async fn person_get(
+    client: &KanidmClient,
+    _person: &Person,
+) -> Result<(TransitionResult, EventRecord), Error> {
+    // Should we measure the time of each call rather than the time with multiple calls?
+    let start = Instant::now();
+    let result = client.idm_person_account_get("idm_admin").await;
+    let end = Instant::now();
+
+    let duration = end.duration_since(start);
+
+    match result {
+        Ok(_) => Ok((
+            TransitionResult::Ok,
+            EventRecord {
+                start,
+                duration,
+                details: EventDetail::PersonGet,
+            },
+        )),
+        Err(client_err) => {
+            debug!(?client_err);
+            Ok((
+                TransitionResult::Error,
+                EventRecord {
+                    start,
+                    duration,
+                    details: EventDetail::Error,
+                },
+            ))
+        }
+    }
+}
+
+pub async fn person_set(
+    client: &KanidmClient,
+    person: &Person,
+) -> Result<(TransitionResult, EventRecord), Error> {
+    // Should we measure the time of each call rather than the time with multiple calls?
```

Comment:
Each call timed individually so they can become their own event records.

Reply:
I think it makes sense to keep it as a single transition and return something like `(TransitionResult, Vec<EventRecord>)` (Actually I came up with another enum to avoid having vecs everywhere). Alternatively we could use two separate transitions but I don't really see the point here as these two calls are basically always done together. In general having the ability to perform multiple calls timed individually could help limit "useless" transitions while still having precise measurements.

Reply:
I guess it depends what we want to measure. I still think any singluar transition should be it's own result and event record. If we have to make multiple calls, we either coalesce them in one transition and start/pause/start/stop the timer. Or we have the State Login -> State Logging In -> State Complete. And then we get three distinct events.

Reply:
Alright so I think we will need a "reauth" state, if we want to have a separate transition for that

Reply:
Do you mean "reauth" as in for priv reauthentication? Or as in to logout and in? Because to logout and in, you'd move back to State::Unauthenticated

Reply:
Yeah it’s about priv reauth, that’s why I wanted to bind it together with the modify operation

Reply:
Oky, need re-auth should be it's own state, and then internally the actor can track it's priv state and if it needs the re-auth or not. It shouldn't be bounh with modify. 

Have you worked with state machines before.?

Reply:
Alright sounds reasonable. 

> Have you worked with state machines before.?

Yeah but I didn't think this through that much have I